### DISCUSSION TITLE
`XmppConnection`  `Capabilities` what is this for?

### DISCUSSION THREAD

**Discussion by ruicaramalho** [vitalyster/SharpXMPP#125] [Q&A] [answered]
Hi.

What's the use of `Capabilities` why does it creates a Node `http://bggg.net.ru/caps` ?

Code from `SharpXMPP.Shared/XmppConnection.cs`

~~~~charp
protected XmppConnection(string ns)
{
    queries = new Dictionary<string, Action<XMPPIq>>();
    _ns = ns;
    Capabilities = new CapabilitiesManager
    {
        Identity = new Identity
        {
            Category = "client",
            IdentityType = "mobile",
            IdentityName = "SharpXMPP"
        },

        Node = "http://bggg.net.ru/caps",
        Features = new List<string>
        {
            Namespaces.DiscoInfo,
            Namespaces.DiscoItems
        }
    };

    IqManager = new IqManager
    {
        PayloadHandlers =
        {
            new InfoHandler(Capabilities),
            new ItemsHandler()
        }
    };
    Iq += IqManager.Handle;
}
~~~~

**Comment by vitalyster** [comment, marked as answer]
This is a default implementation of the [XEP-0115](https://xmpp.org/extensions/xep-0115.html)
`Capabilities` are public, so you can override identity, node, and add features in your extensions

**Reply by ruicaramalho** [reply]
I got it, don't you think **it should be null in the constructor**?  
PS - You are realy quick in answering, thanks.

**Reply by vitalyster** [reply]
Some default implementation is a must for a good XMPP client so we can not disable it

**Reply by vitalyster** [reply]
Most of (if not all) other XMPP libraries provide default capabilities with their node, it is implicit for client to update it with own node when needed
